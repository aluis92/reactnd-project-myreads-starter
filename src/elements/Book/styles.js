import styled, { css } from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

const bookFontSize = css`
    font-size: 0.8em;
`;

export const Container = styled.div`
    width: 140px;
`;

export const BookTitle = styled.h3`
    ${bookFontSize}

    margin-top: 10px;
`;

export const BookAuthors = styled.p`
    ${bookFontSize}

    color: #999;
`;

export const BookTop = styled.div`
    position: relative;
    height: 200px;
    display: flex;
    align-items: flex-end;
`;

export const BookCover = styled.div`
    width: 128px;
    height: 193px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    background: #eee;
`;
