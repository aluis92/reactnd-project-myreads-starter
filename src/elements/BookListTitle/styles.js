import styled from 'styled-components';

export const Container = styled.header`
    padding: 10px 0;
    background: #2e7c31;
    text-align: center;
`;

export const Title = styled.h1`
    font-weight: 400;
    margin: 0;
    color: white;
`;
