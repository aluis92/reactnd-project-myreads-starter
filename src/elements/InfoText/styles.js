import styled from 'styled-components';

export const Container = styled.header`
    padding: 50px;
`;

export const Text = styled.h1`
    text-align: center;
    font-size: 1.5em;
`;
