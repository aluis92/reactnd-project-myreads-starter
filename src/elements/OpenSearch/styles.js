import styled from 'styled-components';
import { Link } from 'react-router-dom';
import addIcon from '../../assets/icons/add.svg';

export const Container = styled.header`
    position: fixed;
    right: 25px;
    bottom: 25px;
`;

export const OpenSearchLink = styled(Link)`
    display: block;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    background: #2e7d32;
    background-image: url(${addIcon});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 28px;
    box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
    font-size: 0;
`;
