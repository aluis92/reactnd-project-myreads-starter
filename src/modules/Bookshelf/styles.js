import styled from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

export const Container = styled.div`
    padding: 0 10px 20px;

    @media (min-width: 600px) {
        .bookshelf {
            padding: 0 20px 40px;
        }
    }
`;

export const BookshelfTitle = styled.h2`
    border-bottom: 1px solid #dedede;
`;

export const BookshelfBooks = styled.div`
    text-align: center;
`;

export const BooksGrid = styled.ol`
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    li {
        padding: 10px 15px;
        text-align: left;
    }
`;
