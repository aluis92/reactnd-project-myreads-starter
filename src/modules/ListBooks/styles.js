import styled from 'styled-components';

export const Container = styled.section`
    position: relative
`;

export const Content = styled.div`
    padding: 0 0 80px;
    flex: 1;
`;
