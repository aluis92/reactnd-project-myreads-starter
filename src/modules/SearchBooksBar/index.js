import React from 'react';
import { func } from 'prop-types';

import {
    Container,
    SearchInput,
    CloseSearch,
    SearchInputWrapper,
} from './styles';

function SearchBooksBar({
    onSearch,
}) {
    return (
        <Container>
            <CloseSearch to="/">Close</CloseSearch>
            <SearchInputWrapper>
                <SearchInput
                    type="text"
                    placeholder="Search by title or author"
                    onChange={e => onSearch(e.target.value)}
                />
            </SearchInputWrapper>
        </Container>
    );
}

SearchBooksBar.propTypes = {
    onSearch: func.isRequired,
};

export default SearchBooksBar;
