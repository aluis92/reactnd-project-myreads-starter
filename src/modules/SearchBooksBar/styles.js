import styled from 'styled-components';
import { Link } from 'react-router-dom';
import backArrow from '../../assets/icons/arrow-back.svg';

export const Container = styled.header`
    padding: 50px 10px 20px;
`;

export const Content = styled.div`
    position: fixed;
    width: 100%;
    top: 0;
    left: 0;
    z-index: 5;
    display: flex;
    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 0 6px rgba(0,0,0,0.23);
`;

export const CloseSearch = styled(Link)`
    display: block;
    top: 20px;
    left: 15px;
    width: 50px;
    height: 53px;
    background: white;
    background-image: url(${backArrow});
    background-position: center;
    background-repeat: no-repeat;
    background-size: 28px;
    font-size: 0;
`;

export const SearchInput = styled.input`
    width: 100%;
    padding: 15px 10px;
    font-size: 1.25em;
    border: none;
    outline: none;
`;

export const SearchInputWrapper = styled.div`
    flex: 1;
    background: #e9e;
`;
