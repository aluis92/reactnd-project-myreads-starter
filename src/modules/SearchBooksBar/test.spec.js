import React from 'react';
import {
    shallow,
} from 'enzyme';

import { SearchInput } from './styles';

import SearchBooksBar from './index';

describe('SearchBooksBar component', () => {
    const defaultSearchBooksBar = {
        onSearch: jest.fn(),
    };

    it('should render without crashing if it has onSearch prop', () => {
        expect(shallow(
            <SearchBooksBar {...defaultSearchBooksBar} />,
        )).toMatchSnapshot();
    });

    it('should call onSearch prop if input value changes', () => {
        const searchBooksBarProps = {
            onSearch: jest.fn(),
        };

        const searchBooksBar = shallow(
            <SearchBooksBar {...searchBooksBarProps} />,
        );

        searchBooksBar
            .find(SearchInput)
            .simulate('change', { target: { value: 'test' } });

        expect(searchBooksBarProps.onSearch).toHaveBeenCalled();
    });
});
