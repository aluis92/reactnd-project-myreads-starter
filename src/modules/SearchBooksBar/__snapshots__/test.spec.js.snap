// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBooksBar component should render without crashing if it has onSearch prop 1`] = `
<styled.header>
  <styled.div>
    <Styled(Link)
      to="/"
    >
      Close
    </Styled(Link)>
    <styled.div>
      <styled.input
        onChange={[Function]}
        placeholder="Search by title or author"
        type="text"
      />
    </styled.div>
  </styled.div>
</styled.header>
`;
